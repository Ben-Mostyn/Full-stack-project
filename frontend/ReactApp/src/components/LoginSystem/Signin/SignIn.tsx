import { useMutation } from "@apollo/client";
import { useState } from "react";
import { LOGIN_USER_MUTATION } from "../../../Graphql/logIn-user";

const SignIn = () => {
    const [formData, setFormData] = useState({ email: '', password: '' });
    const [logInUser, { data, loading, error }] = useMutation(LOGIN_USER_MUTATION);

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setFormData({ ...formData, [e.target.name]: e.target.value });
    }

    const handleSubmit = async (e: React.FormEvent) => {
        e.preventDefault();
        try {
            await logInUser({ variables: formData });
            alert('User logged in lets gooo');
            setFormData({ email: '', password: '' });
        } catch (error) {
            console.error({ name: 'login error', error });
            alert('User failed to log in');
        }
    }

    return (
        <div className="formContainer">
            {data?.logInUser?.user && <p>Welcome back {data.logInUser.user.username}</p>}
            {loading && <p>Loading...</p>}
            {error && <p>Error: {error.message}</p>}

            <h2>Sign in</h2>
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    placeholder="email"
                    name="email"
                    value={formData.email}
                    onChange={handleChange}
                    required
                />
                <input
                    type="password"
                    placeholder="password"
                    name="password"
                    value={formData.password}
                    onChange={handleChange}
                    required
                />
                <button type="submit">Submit</button>
            </form>
        </div>
    );
}

export default SignIn;
