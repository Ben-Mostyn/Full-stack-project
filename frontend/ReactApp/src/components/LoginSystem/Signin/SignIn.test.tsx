import { MockedProvider } from "@apollo/client/testing"
import SignIn from "./SignIn"
import { render, screen } from "@testing-library/react"

describe('',()=>{
    test('', () => {
        const {container} = render(
            <MockedProvider>
                <SignIn />
            </MockedProvider>
        )

        const heading = screen.getByRole('heading', {level: 2});
        expect(heading).toHaveTextContent('Sign in')


        expect(container).toMatchSnapshot();
    })
})