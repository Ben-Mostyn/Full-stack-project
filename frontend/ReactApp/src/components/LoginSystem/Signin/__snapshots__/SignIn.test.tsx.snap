// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` 1`] = `
<div>
  <div
    class="formContainer"
  >
    <h2>
      Sign in
    </h2>
    <form>
      <input
        name="email"
        placeholder="email"
        required=""
        type="text"
        value=""
      />
      <input
        name="password"
        placeholder="password"
        required=""
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
